enhancements/storage/csi-efs-operator.md

```diff
@@ -0,0 +1,335 @@
+  it should be available and set correct security group. See [this KB article](https://access.redhat.com/articles/5025181)
+  for suggested procedure.
+* It offers `SharedVolume` CRD in `aws-efs.managed.openshift.io/v1alpha1` API group to create PV + PVC for an EFS
+  volume. The EFS volume still needs to be created manually, see the previous bullet!
+  * All that `SharedVolume` really does is that it creates PV with proper settings + a PVC. This allows unprivileged
+    users to create PVs. Cluster admins need to be careful to give permissions to use this CRD only to trusted users.
+  * Using `SharedVolume` is purely optional! Cluster admin may create PVs for EFS volumes manually, it's not harder
+    than using `SharedVolume`.
+
+### AWS EFS CSI driver
+
+AWS EFS CSI driver is different than the other CSI drivers we ship. Listing the main differences here for completeness.
+
+* Right now, it does not implement dynamic provisioning. As described above, cluster admin must create the volume in
+  AWS and figure out the networking + security manually.
+  * [Long upstream thread about dynamic provisioning](https://github.com/kubernetes-sigs/aws-efs-csi-driver/issues/6)
+  * Upstream [will offer some sort of dynamic provisioning in the next version](https://github.com/kubernetes-sigs/aws-efs-csi-driver/pull/274)
+    by creating subdirectories (Access Points) in existing EFS volumes as PVs. AWS has a limit of 120 Access Points per
+    EFS volume, i.e. only 120 PVs can be created for a single StorageClass.
+    [Upstream issue](https://github.com/kubernetes-sigs/aws-efs-csi-driver/issues/310).
+  * Proper dynamic provisioning can come in the future, dealing with the networking and security groups is the hardest
+    part there.
+* AWS EFS volumes do not have any size enforcement. They can hold as much data as apps put there and someone will have to
+  pay for the storage. OCP offer standard metrics for amount of data on PVCs, it's up to the cluster admin to create
+  alerts if the consumption gets too high.
+* It uses [efs-utils](https://github.com/aws/efs-utils) to provide encryption in transport, which is not included in
+  RHEL. We will need to package it, either as an image or RPM package.
+
+## Proposal
+High level only, see design details below.
+
+1. Rewrite current aws-efs-operator to use [CSI driver installation functions of library-go](https://github.com/openshift/library-go/tree/master/pkg/operator/csi/csicontrollerset).
+   * This allows us to share well tested code to install CSI drivers and implement bugfixes / new features at a single
+     place.
+   * It will add common features of CSI driver operators which are missing in the current community operator, such as
+     proxy injection and restarting the drivers when cloud credentials change.
+
+2. Update (rewrite) current aws-efs-operator to watch `ClusterCSIDriver` CR named `efs.csi.aws.com`.
```

Comment:
Will aws-efs-operator ever *write* (status, spec, labels, annotations, ...) to the CR? (Concerns about one CRD having multiple "owners".)

Reply:
Yes, it will write status there. The CRD is watched by several operators, but each operator watches its own instance with explicitly defined name and does not touch other CRs.
https://github.com/openshift/api/blob/4b79815405ec40f1d72c3a74bae0ae7da543e435/operator/v1/0000_90_cluster_csi_driver_01_config.crd.yaml#L39

Reply:
Added a note about this.